src/rust/entrystore/src/seg/memcache.rs

```diff
@@ -130,12 +130,28 @@ impl MemcacheStorage for Seg {
         }
     }
 
-    fn incr(&mut self, _key: &[u8], _value: u64) -> Result<u64, MemcacheStorageError> {
-        Err(MemcacheStorageError::NotSupported)
+    fn incr(&mut self, key: &[u8], value: u64) -> Result<u64, MemcacheStorageError> {
+        match self.data.wrapping_add(key, value) {
+            Err(SegError::NotFound) => Err(MemcacheStorageError::NotFound),
+            Err(SegError::NotNumeric) => Err(MemcacheStorageError::Error),
+            Err(_) => Err(MemcacheStorageError::ServerError),
+            Ok(item) => match item.value() {
+                Value::U64(v) => Ok(v),
```

Comment:
nitpicking: I wonder if we should either have the `Ok(item) => ... Value::U64(v)` be the first or last line of this entire code bloc, so the error-handling are visually grouped.

Reply:
Good point. That's more consistent with the other blocks too. Fixed and pushed.